### PR TITLE
Add types for pragma

### PR DESCRIPTION
**Pull request by e6nlaq**
Added type definition file for [pragma](https://www.npmjs.com/package/pragma).

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/pragma/index.d.ts
1. types/pragma/pragma-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 04742957e0faf8de86ae16141b7a8c65909e35a3
</p>

File: types/pragma/index.d.ts
Comment (MysteryBlokHed) [inline types/pragma/index.d.ts:6]: JSDoc types (eg. `{string}`) aren't required for TypeScript files:

```suggestion
```

You can also use the `@param`s to describe each parameter

File: types/pragma/index.d.ts
Comment (MysteryBlokHed) [inline types/pragma/index.d.ts:8]: FYI, the `object` type doesn't refer exclusively to objects like `{}`; it also includes arrays and class instances.

You might want something like this:

```suggestion
declare function pragma(input: string, option?: { parseContent?: (source: string) => any }): Record<string, any>;
```

**Review by MysteryBlokHed — CHANGES_REQUESTED** [review]

**Review by MysteryBlokHed — APPROVED** [review]
LGTM, but will need maintainer approval to get merged

**Review by jakebailey — APPROVED** [review]
Seems fine, though I don't really get why this is being submitted; it's a package which only gets 100 weekly downloads and hasn't been updated in 7 years.

**Comment by e6nlaq** [comment]
> it's a package which only gets 100 weekly downloads and hasn't been updated in 7 years.

You are absolutely right, but I thought this package would be useful in some way.

**Comment by e6nlaq** [comment]
Ready to merge

**Pull request merged**
